lib/config.cpp

```diff
@@ -1310,6 +1305,42 @@ json Config::get_interface_definition(const std::string& interface_name) const {
     return this->interface_definitions.value(interface_name, json());
 }
 
+void Config::populate_module_config_cache() {
+    for (const auto& [module_id, module_name] : this->module_names) {
+        this->module_config_cache[module_id] = ConfigCache();
+        const std::set<std::string> provided_impls = Config::keys(this->manifests.at(module_name).at("provides"));
+        this->interfaces[module_name] = json({});
+        this->module_config_cache[module_name].provides_impl = provided_impls;
```

Comment:
I think this must be 
```suggestion
        this->module_config_cache[module_id].provides_impl = provided_impls;
```

Reply:
fixed, it's actually `module_name` that's needed everywhere